src/main/java/org/alliancegenome/curation_api/services/validation/AlleleValidator.java

```diff
@@ -640,6 +661,16 @@ private List<AlleleGeneAssociation> validateAlleleGeneAssociations(Allele uiEnti
 				}
 			}
 		}
+		if (CollectionUtils.isNotEmpty(dbAssociations)) {
+			for (int ix = 0; ix < dbAssociations.size(); ix++) {
+				AlleleGeneAssociation ga = dbAssociations.get(ix);
+				if(associationsToDelete.contains(ga)){
```

Comment:
This contains is doing a object compare and you are not guaranteed that two identical data contain objects will be equal they may have two different object ids but yet the same object and this contains will always be false.

Reply:
Yeah, that's what I did at first. For some reason using the `uiAssociations.contains(dbAssociation)` up top was always returning null for me.